packages/web-api/src/auth/routes.ts

```diff
@@ -170,6 +197,91 @@ router.get(
   }
 );
 
+/**
+ * Change user password
+ */
+router.post(
+  '/change-password',
+  authRateLimit(5),
+  passport.authenticate('jwt', { session: false }),
+  processRequest({ body: ChangePasswordInputSchema }),
+  async (req, res: Response) => {
+    const { confirmPassword, newPassword, oldPassword } = req.body;
+
+    // check passwords match
+    if (newPassword !== confirmPassword) {
+      throw new Exceptions.BadRequestException('New password and confirmation do not match.');
+    }
+
+    const user = req.user;
+    if (!user) {
+      throw new Exceptions.UnauthorizedException('Invalid credentials.');
+    }
+
+    // Lookup user from DB
+    const userDb = await prisma.user.findUnique({
+      where: { email: user.email },
+      select: {
+        id: true,
+        email: true,
+        password: true,
+        roles: true
+      }
+    });
+
+    if (!userDb) {
+      // This is actually a weird state - log in sentry but reject
+      logSentryMessage(
+        `User who had valid token tried to change password but email didn't exist! Info: ${{ tokenEmail: user.email }}`,
+        'warning'
+      );
+      throw new Exceptions.UnauthorizedException('Invalid credentials.');
+    }
+
+    // Check password against db (latest)
+    const isPasswordValid = await bcryptjs.compare(oldPassword, userDb.password);
+
+    if (!isPasswordValid) {
```

Comment:
`compare` returns `number`, so IMO better to do `!== 0`
Also, the variable name confuses this a bit valid versus matches.
`oldPasswordMatchesDb`

Reply:
compare is Promise<boolean> - leaving as is - however agreew ith varaible name